oshell.sh

```diff
@@ -43,66 +57,108 @@ function start_oci_auth_refresher() {
+  if ! oci session authenticate --profile-name "$profile_name"; then
     echo "OCI authentication failed"
     return 1
   fi
 
   unset OCI_CLI_TENANCY OCI_TENANCY_NAME OCI_COMPARTMENT CID OCI_CLI_REGION
   echo "Setting OCI Profile to ${profile_name}"
   export OCI_CLI_PROFILE=$profile_name
+  set_profile_paths
 
   log_message "Checking for existing refresher process for $OCI_CLI_PROFILE"
-  local refresher_pid=$(find_oci_auth_refresher_process $OCI_CLI_PROFILE)
+  local refresher_pid
+  refresher_pid=$(find_oci_auth_refresher_process "$OCI_CLI_PROFILE")
 
   if [[ -n "$refresher_pid" ]]; then
-    start_oci_auth_refresher $OCI_CLI_PROFILE "true"
+    start_oci_auth_refresher "$OCI_CLI_PROFILE" "true"
   else
-    start_oci_auth_refresher $OCI_CLI_PROFILE "false"
+    start_oci_auth_refresher "$OCI_CLI_PROFILE" "false"
   fi
 
-  oshiv info
+  oci_tenancy_map "$@"
 }
 
 alias ociexit='oci_auth_logout'
 function oci_auth_logout() {
-  if [[ -z "${OCI_CLI_PROFILE}" ]]; then
+  local profile_name="${1:-$OCI_CLI_PROFILE}"
+
+  if [[ -z "${profile_name}" ]]; then
     echo "No active OCI profile found. Nothing to log out from."
     return 0
   fi
 
-  local refresher_pid=$(find_oci_auth_refresher_process $OCI_CLI_PROFILE)
+  # Temporarily set OCI_CLI_PROFILE to the provided profile for path setting
+  local original_profile="$OCI_CLI_PROFILE"
+  export OCI_CLI_PROFILE="$profile_name"
+  set_profile_paths
+
+  # Check if a refresher process exists and terminate it
+  local refresher_pid
+  refresher_pid=$(find_oci_auth_refresher_process "$profile_name")
+  local refresher_terminated=false
+
   if [[ -n "$refresher_pid" ]]; then
-    log_message "Killing refresher process for profile ${OCI_CLI_PROFILE}"
-    kill -9 $refresher_pid
-    echo "expired" > ${HOME}/.oci/sessions/$OCI_CLI_PROFILE/session_status
+    echo "Killing refresher process for profile ${CYAN}${profile_name}${UNSET_FMT}"
+    log_message "Killing refresher process for profile ${profile_name}"
+    kill -9 "$refresher_pid"
+    echo "expired" > "$SESSION_STATUS_FILE"
+    echo "Successfully terminated background refresher for profile: ${CYAN}${profile_name}${UNSET_FMT}"
+    refresher_terminated=true
+  else
+    echo "No background refresher found for profile: ${CYAN}${profile_name}${UNSET_FMT}"
   fi
 
-  oci session terminate
-  if [[ $? -eq 0 ]]; then
-    echo "Successfully logged out from OCI profile: ${CYAN}${OCI_CLI_PROFILE}${UNSET_FMT}"
+  # Always attempt to terminate the session if we're working with the current active profile
+  if [[ "$profile_name" == "$original_profile" ]]; then
+    log_message "Attempting to terminate OCI session for profile ${profile_name}"
+
+    # Capture the output of the command in a variable
+    local terminate_output
+    terminate_output=$(oci session terminate --profile "$profile_name" 2>&1)
+    local terminate_status=$?
+
+    # Log the output
+    log_message "OCI session terminate output: ${terminate_output}"
```

Comment:
I'm seeing errors regarding missing log files around this point in the code:
<img width="1006" alt="image" src="https://github.com/user-attachments/assets/1ed2d627-6272-4ee8-bbeb-0adc3fe006bb" />

Reply:
![Screenshot 2025-07-09 at 2 35 00 PM](https://github.com/user-attachments/assets/68e092b9-58bc-4d59-962f-ad566d98bbf0)